discovery/manager_test.go

```diff
@@ -749,46 +751,50 @@ func verifyPresence(t *testing.T, tSets map[poolKey]map[string]*targetgroup.Grou
 }
 
 func TestTargetSetRecreatesTargetGroupsEveryRun(t *testing.T) {
-	cfg := &config.Config{}
-
-	sOne := `
-scrape_configs:
- - job_name: 'prometheus'
-   static_configs:
-   - targets: ["foo:9090"]
-   - targets: ["bar:9090"]
-`
-	if err := yaml.UnmarshalStrict([]byte(sOne), cfg); err != nil {
-		t.Fatalf("Unable to load YAML config sOne: %s", err)
-	}
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
 	discoveryManager := NewManager(ctx, log.NewNopLogger())
 	discoveryManager.updatert = 100 * time.Millisecond
 	go discoveryManager.Run()
 
 	c := make(map[string]sd_config.ServiceDiscoveryConfig)
-	for _, v := range cfg.ScrapeConfigs {
-		c[v.JobName] = v.ServiceDiscoveryConfig
+	c["prometheus"] = sd_config.ServiceDiscoveryConfig{
+		StaticConfigs: []*targetgroup.Group{
```

Comment:
since the later `StaticConfigs` is just this `[]Group` without the second `Group`, can you define each once and then just reference them when assigning to `c["prometheus"]?

Reply:
When I play with the tests, I will change the config here and there. I don't think it requires to be compact here.
e.g. I may change this one to `foo1:9090` and later one to `foo2:9090`, it's a good idea to duplicate them here.

Reply:
I think we should collapse the target group definitions until there's a need within the tests for them to actually be different.